services-camel/src/test/java/edu/unc/lib/cdr/FulltextProcessorTest.java

```diff
@@ -56,46 +66,70 @@
 	
 	@Mock
 	private Exchange exchange;
+	
 	@Mock
 	private Message message;
+	
+	@Mock
+	private FulltextProcessor fulltext;
 
 	@Before
 	public void init() throws Exception {
 		initMocks(this);
 		processor = new FulltextProcessor(this.repository, slug, fileSuffix);
 		file = File.createTempFile("testFile", "txt");
+		file.deleteOnExit();
 		when(exchange.getIn()).thenReturn(message);
 		PIDs.setRepository(repository);
 		when(repository.getBaseUri()).thenReturn("http://fedora");
 	}
 	
 	@Test
 	public void validTest() throws Exception {
-		BufferedWriter writeFile = new BufferedWriter(new FileWriter(this.file));
-		writeFile.write(testText);
-		writeFile.close();
+		try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+			BufferedWriter writeFile = new BufferedWriter(new FileWriter(this.file));
+			writeFile.write(testText);
+			writeFile.close();
+			
+			when(message.getBody(eq(InputStream.class)))
+					.thenReturn(new ByteArrayInputStream(bos.toByteArray()));
+		}
 		
 		String filePath = this.file.getAbsolutePath().toString();
 		
 		when(message.getHeader(eq(FCREPO_URI)))
-		.thenReturn("http://fedora/test/original_file");
+				.thenReturn("http://fedora/test/original_file");
+		
+		when(message.getHeader(eq(CdrBinaryChecksum)))
+		.thenReturn("1234");
 		
 		when(message.getHeader(eq(CdrBinaryPath)))
 				.thenReturn(filePath);
 		
 		when(message.getHeader(eq(CdrBinaryMimeType)))
-		.thenReturn("plain/text");
+				.thenReturn("plain/text");
+		
+	//	when(message.get)
 		
 		binary = mock(BinaryObject.class);
 		parent = mock(FileObject.class);
 
 		when(repository.getBinary(any(PID.class))).thenReturn(binary);
 		when(binary.getParent()).thenReturn(parent);
+		
+		
 
 		processor.process(exchange);
+		
+		ArgumentCaptor<FulltextProcessor> requestCaptor = ArgumentCaptor.forClass(FulltextProcessor.class);
```

Comment:
fullText isn't a mock (and it shouldn't be since this unit test is testing that class (you are declaring it as a Mock and then assigning a real value in the init method)), so verifying this probably isn't doing anything.  

I can't see my comment anymore, but what you probably should be doing is verify(parent).addDerivative(...) with the argument captor.